buildpacks/ruby/src/steps/get_default_process.rs

```diff
@@ -64,16 +64,33 @@ fn detect_web(gem_list: &GemList, app_path: &Path) -> WebProcess {
     }
 }
 
-fn default_rack() -> Process {
-    ProcessBuilder::new(process_type!("web"), "bundle")
-        .args(["exec", "rackup", "--port", "$PORT", "--host", "0.0.0.0"])
+fn bashify(
+    program: impl Into<String>,
+    args: impl IntoIterator<Item = impl Into<String>>,
+) -> Process {
+    let args = args
+        .into_iter()
+        .map(std::convert::Into::into)
+        .collect::<Vec<String>>()
+        .join(" ");
```

Comment:
This will produce unexpected results when an an arg contains spaces. Maybe https://docs.rs/shell-words/latest/shell_words/ is a good choice to add here. It wouldn't be a big problem when all `Process`es are created literally, but this helper "promises" to convert any program + args into a bash process. I'd argue not properly quoting is a bug.

Reply:
There are some unintended side effects off just slapping that on there:

```
exec rackup --port '$PORT' --host 0.0.0.0
```

In this case I want it to be `"$PORT"` which gets evaluated instead of `'$PORT'` which is a string literal. We ran into this in the hatchet library as well where we ended up needing a way to specify "I actually want this to be evaluated". Maybe an enum:

```rust
enum ShellString {
 Escape(String), // shell escape whatever we pass in
 Quote(String), // ensure contents are wrapped in a doublequote but not shell escaped
 Raw(String), // Do nothing, pass whatever we're given, responsibility is on the person inputting the contents
}
```

Then the API could take an iterator of ShellString and force users to be explicit with how they want their args escaped (or not).